src/views/register/RegisterView.js

```diff
@@ -49,16 +52,74 @@ class RegisterView extends React.Component {
         this.setState(obj);
         this.setState({disabled: !this.validInput()});
     }
-    
+
+    pickImage = async () => {
+        let pickerResult = await ImagePicker.launchImageLibraryAsync({
+            allowsEditing: true,
+            aspect: [4, 3],
+            quality: 0.8,
+            base64: true,
+            mediaTypes: ImagePicker.MediaTypeOptions.Images 
+        });
+        
+        if (!pickerResult.cancelled) {
+            console.log("avatarUrl:", pickerResult.uri)
+            this.setState({ avatarUrl: pickerResult.uri, disabled: false});
+            let base64Img = `data:image/jpg;base64,${pickerResult.base64}`
+            this.base64Img = base64Img;
+        }
+    };
+
+    updateAvatar = async () =>{
+        this.setState({spinner: true})
+        try {
```

Comment:
There is too much logic here.
React components shouldn't create the ajax request, they should use an external service for it. exactly what `pickImage` doing great :+1:

Reply:
I don't like the mixed usage of `async/await` with `then`.
I think the code will look much better if you replace all your `then` with `await`